x-pack/platform/plugins/shared/observability_ai_assistant/server/routes/chat/route.ts

```diff
@@ -269,6 +268,16 @@ async function chatComplete(
     scopes,
   });
 
+  const userInstructionsWithId: Instruction[] | undefined = userInstructions?.map(
+    (userInstruction) =>
+      typeof userInstruction === 'string'
+        ? {
+            text: userInstruction,
+            id: v4(),
+          }
+        : userInstruction
+  );
```

Comment:
nit: let's flip it around, so it only becomes a "user instruction" once we have added the id. 
```suggestion
  const userInstructions: Instruction[] | undefined = userInstructionsOrStrings?.map(
    (userInstructionOrString) =>
      typeof userInstructionOrString === 'string'
        ? {
            text: userInstructionOrString,
            id: v4(),
          }
        : userInstructionOrString
  );
```